packages/web-app-files/src/components/SideBar/SideBar.vue

```diff
@@ -234,71 +134,13 @@ export default {
       this.$set(this.selectedFile, 'starred', starred)
     }
   },
-
   async created() {
     if (!this.areMultipleSelected) {
       await this.fetchFileInfo()
     }
-    this.$nextTick(this.initVisibilityObserver)
-  },
-
-  beforeDestroy() {
-    visibilityObserver.disconnect()
-    hiddenObserver.disconnect()
+    this.$nextTick(this.$refs.sidebar.initVisibilityObserver())
```

Comment:
Can we do this better?

We need to wait for `fetchFileInfo()` to finish before the list of panels is known, that's why we can only initialize the observer then... 
If we need to do it like this, should we maybe invoke `initVisibilityObserver()` by default in `created()` of the SideBar component in `web-pkg` and add an option to opt-out and then do it yourself here? I'd not expect I'd have to call this function usually, so making it mandatory by default seems weird ...